module/spl/spl-kmem.c

```diff
@@ -133,6 +133,74 @@ strfree(char *str)
 }
 EXPORT_SYMBOL(strfree);
 
+/* kernel compat for <4.13 */
+#ifndef __GFP_RETRY_MAYFAIL
+#define	__GFP_RETRY_MAYFAIL	__GFP_REPEAT
+#endif
+
+void *
+spl_kvmalloc(size_t size, gfp_t lflags)
+{
+#ifdef HAVE_KVMALLOC
+	/*
+	 * GFP_KERNEL allocations can safely use kvmalloc which may
+	 * improve performance by avoiding a) high latency caused by
+	 * vmalloc's on-access allocation, b) performance loss due to
+	 * MMU memory address mapping and c) vmalloc locking overhead.
+	 * This has the side-effect that the slab statistics will
+	 * incorrectly report this as a vmem allocation, but that is
+	 * purely cosmetic.
+	 */
+	if ((lflags & GFP_KERNEL) == GFP_KERNEL)
+		return (kvmalloc(size, lflags));
```

Comment:
If we are going to implement the same thing for kernels that lack `kvmalloc()`, what is the benefit of using it when it is available? We could have a uniform behavior across all kernels, one less autotools check and fewer lines of code by not using `kvmalloc()`.

Reply:
No, we would need even more autotools checks as we had to differntiate between kernel <4.12, =4.12, =4.13 and >4.13. This was discussed already here https://github.com/zfsonlinux/zfs/pull/9034#discussion_r305522509